### DISCUSSION TITLE
Manage app notifications

### DISCUSSION THREAD

**Discussion by taorepoara** [lenra-io/lenra_cli#112] [General]
Create a notification system to let the app creators handle notifications for there users.

A notification will be defined with a scope that work like a path. This scope will permit the users to disable notifications at any level of the scope and re-enable it in a lower path. 

For example in a messaging app, if I create a notification with this scope `conversation.my-group.message.Paul` when my friend Paul send a message through the `my-group` conversation, by default the user will receive it. If he disable notifications for the scope `conversation.my-group`, he won't receive it anymore and if he enable back the scope `conversation.my-group.message.Paul` he will only will receive notifications when Paul send a message, but not any other (other users message or new user in the conversation).

This let the user handle notifications as he like. We can imagine disabling notifications during a given time just like WhatsApp does.

**Comment by Nesqwik** [comment]
Good idea. This scope system should give the dev enough granularity to do almost everything and keep it easy to use for simple apps.
